services/kafka/service.go

```diff
@@ -60,7 +150,14 @@ func (c *Cluster) writer(topic string) (*kafka.Writer, error) {
 				return nil, err
 			}
 			wc.Topic = topic
-			w = kafka.NewWriter(wc)
+			kw := kafka.NewWriter(wc)
+			// Create new writer
+			w = &writer{
+				kafka:   kw,
+				cluster: c.cfg.ID,
+				topic:   topic,
+			}
+			w.Open()
```

Comment:
do we need a routine to cleanup and close all the writers at some point?

Reply:
Writers already had logic to close them. This change just creates the companion Open method since now the initialization is more complex than creating the type.